crates/holochain_state/src/source_chain.rs

```diff
@@ -510,6 +516,12 @@ impl SourceChain {
             let entries = scratch.drain_entries().collect::<Vec<_>>();
             SourceChainResult::Ok((headers, ops, entries))
         })?;
+        let mut ops_to_self_publish = HashSet::with_capacity(ops.len());
+        for op in &ops {
+            if network.authority_for_hash(op.0.dht_basis().clone()).await? {
```

Comment:
I think it might be worth the complexity to join_all these so they can run in parallel instead of serially.

As it is, the cpu will have to switch off between the network ghost_actor task and this one, where, if they are all sent at once, ghost_actor internally does a [ready_chunks](https://docs.rs/futures/0.3.16/futures/stream/trait.StreamExt.html#method.ready_chunks) so the cpu can stay on that code until complete.

Reply:
Maybe it makes more sense to just make authority for has take a vec? So we can batch the call?

Reply:
Sure, I like that too :+1: